src/accountsdb/db.zig

```diff
@@ -372,13 +413,33 @@ pub const AccountsDB = struct {
                 std.debug.panic("failed to *open* AccountsFile {s}: {s}\n", .{ file_name, @errorName(err) });
             };
 
-            // validate and count here for prealloc
-            self.account_index.validateAccountFile(&accounts_file, bin_counts, refs_ptr) catch |err| {
+            const reference_slice_start = references.items.len;
+            self.account_index.validateAccountFile(&accounts_file, bin_counts, &references) catch |err| {
+                if (err == error.OutOfReferenceMemory) {
+                    // TODO: support retry - panic for now
+                    std.debug.panic(
```

Comment:
my thoughts were this function should only be run at startup and these errors are un-recoverable so we should just panic 

changed to errors though since i agree with you - b144ae7